trestle/utils/fs.py

```diff
@@ -447,3 +447,31 @@ def allowed_task_name(name: str) -> bool:
         logger.error('Task name cannot contain __global__')
         return False
     return True
+
+
+def relative_resolve(candidate: pathlib.Path, cwd: pathlib.Path) -> pathlib.Path:
+    """Resolve a candidate file path relative to a provided cwd.
+
+    This is to circumvent bad behaviour for resolve on windows platforms where the path must exist.
+
+    If a relative dir is passed it presumes the directory is relative to the PROVIDED cwd.
+    If relative expansions exist (e.g. ../) the final result must still be within the cwd.
+
+    If an absolute path is provided it tests whether the path is within the cwd or not.
+
+    """
+    # Expand user first if applicable.
+    candidate = candidate.expanduser()
+
+    if not cwd.is_absolute():
+        raise TrestleError('Must provide absolute cwd.')
```

Comment:
This error msg might need more descriptive info if user sees it.

Reply:
Done.